pkg/apps/alert/latest/alertpvc_test.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright (C) 2019 Synopsys, Inc.
+
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements. See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownershia. The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied. See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package alert
+
+import (
+	"reflect"
+	"testing"
+
+	alertapi "github.com/blackducksoftware/synopsys-operator/pkg/api/alert/v1"
+	"github.com/blackducksoftware/synopsys-operator/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestSpecConfig_getAlertPersistentVolumeClaim(t *testing.T) {
+	tests := []struct {
+		name          string
+		altSpecConfig *SpecConfig
+		wantErr       bool
+	}{
+		{
+			name: "base test",
+			altSpecConfig: &SpecConfig{
+				alert: &alertapi.Alert{
+					Spec: alertapi.AlertSpec{
+						Namespace:       "alert",
+						PVCSize:         "1G",
+						PVCStorageClass: "standard",
+					},
+				},
+				isClusterScope: false,
+			},
+			wantErr: false,
+		},
+		{
+			name: "invalid size",
+			altSpecConfig: &SpecConfig{
+				alert: &alertapi.Alert{
+					Spec: alertapi.AlertSpec{
+						Namespace:       "alert",
+						PVCSize:         "1GB",
+						PVCStorageClass: "standard",
+					},
+				},
+				isClusterScope: false,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.altSpecConfig.getAlertPersistentVolumeClaim()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("SpecConfig.getAlertPersistentVolumeClaim() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if err == nil {
+				size, _ := resource.ParseQuantity(tt.altSpecConfig.alert.Spec.PVCSize)
+				want := &v1.PersistentVolumeClaim{
```

Comment:
Move the want into a field in the testcase struct.  More easily allows the input and output to be properly associated and modified if needed.

Reply:
Yes and I've done that previously in other unit tests but the reason I'm doing it this way here is so we don't have hundreds of lines of code with static instants (which would be hard to maintain).  This way the want is configurable to the input.